app/services/users/geo_search.rb

```diff
@@ -112,15 +112,23 @@ def available_motifs_from_attributed_agents_arel
   end
 
   def available_motifs_base
-    @available_motifs_base ||= Motif.where(id: individual_motifs + collective_motifs).joins(:organisation)
+    @available_motifs_base ||= Motif.where(id: individual_motifs_ids + collective_motifs_ids).joins(:organisation)
   end
 
   def individual_motifs
-    @individual_motifs ||= Motif.active.individuel.joins(:plage_ouvertures).distinct
+    @individual_motifs ||= Motif.where(id: individual_motifs_ids)
```

Comment:
À ce que je vois nous avons 2 usages de `individual_motifs` : 
- celui dans `available_motifs_base`, qui a besoin de faire 2 sous-requêtes car les jointures sont incompatibles
- celui dans `individual_motifs`  ([ici](https://github.com/betagouv/rdv-solidarites.fr/blob/b4ea5ec3245166f348d4a6f7a854dbba8dd0fd2c/app/services/users/geo_search.rb#L134)), qui lui n'a pas besoin de faire de sous-requête, car il prend soin d'ajouter des filtres précis (sur une organisation et un agent). Il n'est donc pas nécessaire que cet usage fasse indirectement appel à `individual_motifs_ids` (qui fait un appel supplémentaire).

Ce que je te propose donc, c'est que `individual_motifs` ne change pas, mais que `available_motifs_base` prenne une forme de ce genre : 

```ruby
  def available_motifs_base
    # Nous devons passer les IDs car les jointures des 2 sous-requêtes sont incompatibles
    individual_and_collectifs_motif_ids = individual_motifs.select(:id).ids + collective_motifs.select(:id).ids
    @available_motifs_base ||= Motif.where(id: individual_and_collectifs_motif_ids).joins(:organisation)
  end
```

Qu'en penses-tu ? N'hésites pas à venir me voir pour en discuter. :wink:

Reply:
Bien vu! Je fais la modif

Reply:
Fixed in https://github.com/betagouv/rdv-solidarites.fr/pull/3511/commits/906f066051446d6b2110326d1413f871e2fcc8b4